usr/share/rear/output/PXE/default/801_copy_to_http.sh

```diff
@@ -0,0 +1,55 @@
+# 801_copy_to_http.sh
+#
+# copy kernel and initrd to HTTP server for Relax-and-Recover
+#
+# This file is part of Relax-and-Recover, licensed under the GNU General
+# Public License. Refer to the included COPYING for full text of license.
+
+# Nothing to do here when PXE_HTTP_UPLOAD_URL is not set
+if [[ -z "$PXE_HTTP_UPLOAD_URL" ]] ; then
+    LogPrint "PXE_HTTP_UPLOAD_URL not set, skipping copy"
+    return
+fi
+
+# When PXE_HTTP_UPLOAD_URL equals PXE_TFTP_UPLOAD_URL then the files have
+# already been uploaded by 800_copy_to_tftp.sh and we don't need to do
+# anything here.
+if [[ "$PXE_HTTP_UPLOAD_URL" = "$PXE_TFTP_UPLOAD_URL" ]] ; then
+    LogPrint "PXE_HTTP_UPLOAD_URL = PXE_TFTP_UPLOAD_URL, skipping copy to PXE_HTTP_UPLOAD_URL"
+    return
+fi
+
+# E.g. PXE_HTTP_UPLOAD_URL=nfs://server/export/nfs/www
+local scheme=$( url_scheme $PXE_HTTP_UPLOAD_URL )
+
+# We need filesystem access to the destination (schemes like ftp:// are not supported)
+if ! scheme_supports_filesystem $scheme ; then
+    Error "Scheme $scheme from PXE_HTTP_UPLOAD_URL not supported, use a scheme that supports mounting (like nfs: )"
+fi
+
+mount_url $PXE_HTTP_UPLOAD_URL $BUILD_DIR/httpbootfs $BACKUP_OPTIONS
+# However, we copy under $OUTPUT_PREFIX_PXE directory (usually HOSTNAME) to have different clients on one pxe server
+pxe_http_local_path=$BUILD_DIR/httpbootfs
+# mode must readable for others for pxe and we copy under the client HOSTNAME (=OUTPUT_PREFIX_PXE)
+mkdir -m 755 -p $v "$BUILD_DIR/httpbootfs/$OUTPUT_PREFIX_PXE" >&2
+StopIfError "Could not mkdir '$BUILD_DIR/httpbootfs/$OUTPUT_PREFIX_PXE'"
```

Comment:
@jsmeix 
This was copied from `800_copy_to_tftp.sh`. What do you suggest instead? I will then fix it in both.

Reply:
Just use bash directly like
```
mkdir $v -m 755 -p "$BUILD_DIR/httpbootfs/$OUTPUT_PREFIX_PXE" || Error "Could not mkdir '$BUILD_DIR/httpbootfs/$OUTPUT_PREFIX_PXE'"
```
cf. my examples in my comment to "the ...IfError functions".

By the way:

I moved '$v' directly after the command because I find
```
COMMAND $v -option1 -option2 further arguments
```
easier to understand what that '$v' is compared to somewhat hidden
```
COMMAND -option1 -option2 $v further arguments
```

I removed stdout/stderr redirection because
in general this is no longer needed since
https://github.com/rear/rear/issues/2416
except exceptions e.g. when stdout/stderr messages
should be suppressed in any case because they cause
more confusion than helping to debug an issue.
FYI see also
https://github.com/rear/rear/issues/1395

Reply:
By the way:

In general avoid
```
if ! COMMAND ; then ... ; fi
```
becaue '!' changes (inverts) '$?' so
```
if ! COMMAND ; then Error "COMMAND failed with $?" ; fi
```
won't work.

Only when you know you don't need '$?'
you may use `if ! ...` like
```
if ! COMMAND ; then Error "COMMAND failed" ; fi
```
which works until some contributer (correctly) thinks
the exit code would help the user to debug issues
and "just improves" that to
```
if ! COMMAND ; then Error "COMMAND failed with $?" ; fi
```
;-)

Reply:
> because '!' changes/inverts '$?'

good point, I have not seen https://github.com/rear/rear/pull/2985#issuecomment-1545455356 , so I had to discover the same thing myself the hard way: https://github.com/rear/rear/pull/3102/files#diff-0d218219d2a57ecac253bdbab9e27bfd8d6cd4222e103b87f912b8e48f725149R45

Reply:
Fixed with 326cd4a5d5713aaa59e901b52534222b18c908c0